corehq/sql_db/shard_data_management.py

```diff
@@ -99,6 +99,43 @@ def get_count_of_unmatched_models_by_shard(database, model):
     return results
 
 
+def delete_unmatched_shard_data(database, model):
+    """
+    Deletes any `model` data residing on shards that aren't expected to live in `database`.
+
+    Returns number of records deleted
+    """
+    field_type = model._meta.get_field(model.partition_attr)
+    if isinstance(field_type, UUIDField):
```

Comment:
This is currently copied from a below function. It should be pulled out to a separate function